include/soralog/event.hpp

```diff
@@ -35,16 +38,40 @@ namespace soralog {
      * @param level of event
      * @param format and @param args defines message of event
      */
-    template <typename... Args>
+    template <typename ThreadFlag, typename... Args>
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,hicpp-member-init)
-    Event(std::string_view name, Level level, std::string_view format,
-          const Args &... args)
+    Event(std::string_view name, ThreadFlag thread_flag, Level level,
+          std::string_view format, const Args &... args)
         : timestamp_(std::chrono::system_clock::now()), level_(level) {
+      switch (thread_flag) {
+        case ThreadFlag::NAME:
+          util::getThreadName(thread_name_);
+          thread_name_[thread_name_.size() - 1] = 0;  // NOLINT
+          while (thread_name_[thread_name_size_++])   // NOLINT
```

Comment:
Why not use strlen? It took me a while to realize that this loop just counts the size of thread_name_(is it?), and while I'm not the smartest programmer on Earth, I believe this might confuse some other people, too.

Reply:
It's partcle of experimental code =) Thanks for your notification.